### PR TITLE
[CALCITE-3580] Oracle EXTRACT(XML) Function Support

### PR DESCRIPTION
**Pull request by ritesh-kapoor**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/util/SimpleNamespaceContext.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+
+public class SimpleNamespaceContext implements NamespaceContext {
+
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/util/SimpleNamespaceContext.java]: Give some java doc about what is this class used for.

File: core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java [hunk 2]
```diff
@@ -231,6 +231,11 @@ public Consistency getConsistency() {
   public static final FamilyOperandTypeChecker STRING_STRING_STRING =
       family(SqlTypeFamily.STRING, SqlTypeFamily.STRING, SqlTypeFamily.STRING);
 
+  public static final SqlSingleOperandTypeChecker STRING_STRING_OPTIONAL_STRING =
+      family(ImmutableList.of(SqlTypeFamily.STRING, SqlTypeFamily.STRING, SqlTypeFamily.STRING),
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java]: The type should be `FamilyOperandTypeChecker`.

File: site/_docs/reference.md [hunk 3]
```diff
@@ -2281,6 +2281,7 @@ semantics.
 | m | DAYNAME(datetime)                              | Returns the name, in the connection's locale, of the weekday in *datetime*; for example, it returns '星期日' for both DATE '2020-02-10' and TIMESTAMP '2020-02-10 10:10:10'
 | o | DECODE(value, value1, result1 [, valueN, resultN ]* [, default ]) | Compares *value* to each *valueN* value one by one; if *value* is equal to a *valueN*, returns the corresponding *resultN*, else returns *default*, or NULL if *default* is not specified
 | p | DIFFERENCE(string, string)                     | Returns a measure of the similarity of two strings, namely the number of character positions that their `SOUNDEX` values have in common: 4 if the `SOUNDEX` values are same and 0 if the `SOUNDEX` values are totally different
+| o | EXTRACT(xml, xpath, [, namespace ])            | Returns the xml fragment of the element or elements matched by the XPath expression.
 | m | EXTRACTVALUE(xml, xpathExpr))                  | Returns the text of the first text node which is a child of the element or elements matched by the XPath expression.
```
Comment (danny0405) [inline site/_docs/reference.md]: Also give a doc about the optional argument `namespace`.

File: core/src/main/java/org/apache/calcite/util/SimpleNamespaceContext.java (hunk 1 above)
Comment (chunweilei) [reply core/src/main/java/org/apache/calcite/util/SimpleNamespaceContext.java]: +1.

File: core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java (hunk 2 above)
Comment (ritesh-kapoor) [reply core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java]: Previously referred NUMERIC_OPTIONAL_INTEGER. I have done the changes :)

File: site/_docs/reference.md (hunk 3 above)
Comment (ritesh-kapoor) [reply site/_docs/reference.md]: Done :)

File: core/src/main/java/org/apache/calcite/util/SimpleNamespaceContext.java (hunk 1 above)
Comment (ritesh-kapoor) [reply core/src/main/java/org/apache/calcite/util/SimpleNamespaceContext.java]: Done :)

**Pull request closed without merging**